components/shared/BalanceInput.vue

```diff
@@ -37,108 +37,112 @@
 import { units as defaultUnits } from '@/params/constants'
 import { Unit } from '@/params/types'
-import { Debounce } from 'vue-debounce-decorator'
-import ChainMixin from '@/utils/mixins/chainMixin'
+import { useDebounceFn } from '@vueuse/core'
 import { NeoField, NeoInput, NeoSelect } from '@kodadot1/brick'
 
-@Component({
-  components: {
-    NeoField,
-    NeoInput,
-    NeoSelect,
+const { $t } = useNuxtApp()
+const props = defineProps({
+  value: {
+    type: Number,
+    default: 0,
+  },
+  label: {
+    type: String,
+    default: 'amount',
+  },
+  calculate: {
+    type: Boolean,
+    default: true,
+  },
+  expanded: Boolean,
+  step: {
+    type: Number,
+    default: 0.001,
+  },
+  min: {
+    type: Number,
+    default: 0,
+  },
+  max: {
+    type: Number,
+    default: Number.MAX_SAFE_INTEGER,
+  },
+  required: {
+    type: Boolean,
+    default: false,
+  },
+  hasToLargerThanZero: {
+    type: Boolean,
+    default: false,
   },
 })
-export default class BalanceInput extends mixins(ChainMixin) {
-  @Prop({ type: Number, default: 0 }) value!: number
-  @Prop({ default: 'amount' }) public label!: string
-  @Prop({ default: true }) public calculate!: boolean
-  @Prop(Boolean) public expanded!: boolean
-  @Prop({ default: 0.001 }) public step!: number
-  @Prop(Number) public min!: number
-  @Prop({ type: Number, default: Number.MAX_SAFE_INTEGER }) public max!: number
-  @Prop({ type: Boolean, default: false }) public required!: boolean
-  @Prop({ type: Boolean, default: false }) public hasToLargerThanZero!: boolean
-  protected checkZeroFailed = false
-  protected units: Unit[] = defaultUnits
-  private selectedUnit = 1
-  private internalValue = this.value || 0
 
-  get minWithUnit(): number {
-    return this.min / this.selectedUnit
-  }
-
-  get maxWithUnit(): number {
-    return this.max / this.selectedUnit
-  }
+const emits = defineEmits(['input'])
+const selectedUnit = ref(1)
+const internalValue = ref(props.value || 0)
+const checkZeroFailed = ref(false)
+const { decimals, unit: chainUnit } = useChain()
 
-  @Watch('value') onValueChange(newValue) {
-    this.internalValue = newValue
+const mapper = (unit: Unit) => {
+  if (unit.name === '-') {
+    return { ...unit, name: chainUnit.value }
   }
+  return unit
+}
 
-  @Ref('balance') readonly balance
-
-  get inputValue(): number {
-    return this.internalValue
-  }
+const units = ref<Unit[]>(defaultUnits.map(mapper))
 
-  set inputValue(value: number) {
-    this.handleInput(value)
-  }
+const minWithUnit = computed(() => props.min / selectedUnit.value)
+const maxWithUnit = computed(() => props.max / selectedUnit.value)
 
-  public focusInput(): void {
-    this.balance?.focus()
+watch(
+  () => props.value,
+  (newValue) => {
+    internalValue.value = newValue
   }
+)
 
-  formatSelectedValue(value: number): string {
-    return value ? String(value * 10 ** this.decimals * this.selectedUnit) : '0'
-  }
+const balance = ref(null)
+const inputValue = computed({
```

Comment:
`useVmodel`

Reply:
Hello, but we're not editing anything here. We are handling the set with handleInput. 

Please let me know how you'd want me to use useVModel here

Thank you